[Docs] Release process updates

jonatack opened the pull request:
Updates to `release-notes.md`:

- Write an introduction explaining how to use `release-notes.md` as a template for the release notes draft wiki for collaborative editing at https://github.com/bitcoin-core/bitcoin-devwiki/wiki, as seen for the 0.17.0 and 0.18.0 releases.

Updates to `release-process.md`:

- Create a release notes draft wiki at https://github.com/bitcoin-core/bitcoin-devwiki/wiki.

- As per http://www.erisian.com.au/bitcoin-core-dev/log-2019-03-28.html#l-342, for the period during which the release notes are being edited on the wiki, the version on the branch should be wiped and replaced with a link to the wiki which should be used for all announcements until final.

- Before -final, remove the "Needs release note" label from relevant PRs/issues and merge the release notes from the wiki into the branch.

- Create a pinned meta-issue dedicated to testing the release candidate and communicate it in release announcements where useful. The former is done in practice (e.g. https://github.com/bitcoin/bitcoin/issues/15555, https://github.com/bitcoin/bitcoin/issues/14902) and the latter addresses the discussion here yesterday: https://x0f.org/web/statuses/101753569204220416.

- Reorganise the headers in the Branch Updates section.

- Update the version numbers in the examples.

- Adapt and merge in the updates in https://github.com/bitcoin/bitcoin/pull/15692.

jonatack committed:
[docs] Update release-process.md

- Create a release notes draft wiki for collaborative editing at https://github.com/bitcoin-core/bitcoin-devwiki/wiki as seen for releases 0.17.0 and 0.18.0.

- As per http://www.erisian.com.au/bitcoin-core-dev/log-2019-03-28.html#l-342, for the period during which the notes are being edited on the wiki, the version on the branch should be wiped and replaced with a link to the wiki which should be used for all announcements until final.

- Before final, remove the "Needs release note" label from relevant PRs/issues and merge the release notes from the wiki into the branch.

- Create a pinned meta-issue dedicated to testing the release candidate and communicate it in release announcements where useful. The former is done in practice (e.g. #15555, #14902) and the latter addresses the discussion here: https://x0f.org/web/statuses/101753569204220416.

- Adapt and merge the updates in https://github.com/bitcoin/bitcoin/pull/15692.

- Update the version numbers in all the examples.

- Reorganise the headers in the Branch Updates section.

MarcoFalke commented on doc/release-process.md:
```diff
@@ -325,6 +327,8 @@ bitcoin.org (see below for bitcoin.org update instructions).
 
       - Create a [new GitHub release](https://github.com/bitcoin/bitcoin/releases/new) with a link to the archived release notes.
 
+      - Create an issue dedicated to testing the release and provide the link to it in the release announcements where useful
```
I don't like having a single issue/thread for all issues with a release. It could be a meta issue, but all bug reports should go into their own issue.

jonatack replied:
Thank you for reviewing. Prefer re-word (e.g. "meta issue" or "issue for release ACKs") or remove?

jonatack replied:
Updated.

fanquake commented on doc/release-process.md:
```diff
@@ -325,6 +327,8 @@ bitcoin.org (see below for bitcoin.org update instructions).
 
       - Create a [new GitHub release](https://github.com/bitcoin/bitcoin/releases/new) with a link to the archived release notes.
 
+      - Create an pinned meta-issue for testing the release and provide the link to it in the release announcements where useful
```
nit: a pinned meta-issue

fanquake commented:
utACK. Please squash your changes.

jonatack commented:
Updated and squashed. Thanks for reviewing :+1:

fanquake commented:
utACK ff41428

laanwj commented on doc/release-process.md:
```diff
@@ -42,6 +42,8 @@ Check out the source code in the following directory hierarchy.
 
 ### Bitcoin maintainers/release engineers, suggestion for writing release notes
 
+Open a release notes draft for collaborative editing at https://github.com/bitcoin-core/bitcoin-devwiki/wiki
```
Good to mention.
Though if you do, I think it needs to also mention merging this wiki back into the branch just before `-final`

fanquake commented:
Could you also mention somewhere that all PRs/issues should have the `"Needs release note"` label removed after `-final`.

jonatack commented:
Added both suggestions "Before the final release" instead of before and after

jonatack commented:
Also added https://github.com/bitcoin/bitcoin/issues/15555 as an example for the meta issue because it now mentions:

"Let us know which version you tested on which operating system. If you find an issue, please search Github for known issues first and then open a new Github issue."

jonatack commented:
Updated as per core dev IRC discussion http://www.erisian.com.au/bitcoin-core-dev/log-2019-03-28.html#l-342:

"For the period during which the notes are being edited on the wiki, the version on the branch should be wiped and replaced with a link to the wiki which should be used for all announcements until final."

MarcoFalke commented on doc/release-process.md:
```diff
@@ -40,23 +40,33 @@ Check out the source code in the following directory hierarchy.
-Write release notes. git shortlog helps a lot, for example:
+Open a draft of the release notes for collaborative editing at https://github.com/bitcoin-core/bitcoin-devwiki/wiki.
+
+For the period during which the notes are being edited on the wiki, the version on the branch should be wiped and replaced with a link to the wiki which should be used for all announcements until `-final`.
```
The steps to be done pre-branch-off should be mentioned above, no?

See the commit in #15692, which you are welcome to steal.

jonatack replied:
Ok :+1:

jonatack replied:
Stolen, adapted/merged, rebased.

MarcoFalke commented on doc/release-notes.md:
```diff
@@ -1,5 +1,4 @@
-(note: this is a temporary file, to be added-to by anybody, and moved to
-release-notes at release time)
+*(Note: this is a temporary file to be added-to by anybody. See [/doc/developer-notes.md#release-notes](/doc/developer-notes.md#release-notes) for the process. After branching off for a major version release of Bitcoin Core, release notes are drafted in a collaborative wiki at https://github.com/bitcoin-core/bitcoin-devwiki/wiki/*version*-Release-Notes-Draft). Before the final release, the notes  ...
```
Could make this multiline?

```suggestion
*This is a temporary file to be added-to by anybody. See
[/doc/developer-notes.md#release-notes](/doc/developer-notes.md#release-notes)
for the process.*

*After branching off for a major version release of Bitcoin Core, release notes
are drafted in a collaborative wiki at*
https://github.com/bitcoin-core/bitcoin-devwiki/wiki/*version*-Release-Notes-Draft.
*Before the final release, the notes need to be moved back to this git
repository.*
```

jonatack replied:
Done, and used your changes to the writing. Also made the wiki link point to a functional url and did another writing pass.

jonatack commented:
rebased

MarcoFalke commented on doc/release-process.md:
```diff
@@ -25,13 +25,16 @@ Before every major release:
   - set `CLIENT_VERSION_REVISION` to `0`
   - set `CLIENT_VERSION_IS_RELEASE` to `true`
+* Before branching off the release, clear the release notes and move them to the wiki (see "Write the release notes" below).
+* After branching off the release:
```
Instead of nested lists with `*`, could use some headings? I.e.

````
## Updates

### Before every release candidate
...

### Before every major release

#### Before branch-off

...

#### After branch off (on master)

#### After branch off (on the major release branch)

## Building

### First time/New builders

jonatack replied:
Done.

MarcoFalke commented on doc/release-process.md:
```diff
@@ -45,23 +48,33 @@ Check out the source code in the following directory hierarchy.
 
     git tag -s v(new version, e.g. 0.8.0)
 
+Before the final release:
```
Those bullet points should be moved up to the "Branch updates" section? You could link to the "writing release notes" from there

jonatack replied:
Done

MarcoFalke commented on doc/release-process.md:
```diff
@@ -328,7 +341,9 @@ bitcoin.org (see below for bitcoin.org update instructions).
-      - Create a [new GitHub release](https://github.com/bitcoin/bitcoin/releases/new) with a link to the archived release notes.
+      - Create a [new GitHub release](https://github.com/bitcoin/bitcoin/releases/new) with a link to the archived release notes
+
+      - Create a pinned meta-issue for testing the release (see [here](https://github.com/bitcoin/bitcoin/issues/15555) for an example) and provide a link to it in the release announcements where useful
```
This is not done *after* the release, but *before*. Also, I don't like the idea of accumulating bugs of completely different kinds in a single issue. This makes it harder to track. Idk I am ~0 on this

jonatack replied:
Changed "for testing the release" to "for testing the release candidate".

MarcoFalke commented:
ACK beside nits. Thanks for sticking with this

jonatack commented on doc/release-process.md:
```diff
@@ -1,38 +1,59 @@
+
+#### Before final release
+
+- Merge the release notes from the wiki into the branch.
```
Does this line conflict with the instructions in line 351 below? 

"Archive the release notes for the new version to `doc/release-notes/` (branch `master` and branch of the release)"

jonatack commented:
Thanks for reviewing @MarcoFalke. Kept the new commits separate for now, will squash if these changes are all good.

MarcoFalke commented on doc/release-process.md:
```diff
@@ -1,38 +1,59 @@
-* Update `src/chainparams.cpp` defaultAssumeValid with information from the getblockhash rpc.
+### Before every major and minor release
+
+- Update [bips.md](bips.md) to account for changes since the last release.
```
Could you please keep the previous formatting of `*` to minimize the diff and make review easier?

jonatack replied:
Done.

MarcoFalke commented:
ACK

MarcoFalke commented:
Please squash your commits according to https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits

jonatack committed:
[docs] Update release-notes.md

- Write an introduction explaining how to use the release-notes.md template.

- Adapt and merge the updates in https://github.com/bitcoin/bitcoin/pull/15692.

jonatack commented:
Squashed down to 2 logical commits with updated commit messages, and updated the PR description.
